components/canvas/vello_backend.rs

```diff
@@ -0,0 +1,745 @@
+    device: Device,
+    queue: Queue,
+    renderer: Rc<RefCell<vello::Renderer>>,
+    scene: vello::Scene,
+    transform: kurbo::Affine,
+    size: Size2D<u32>,
+}
+
+fn options() -> vello::RendererOptions {
+    vello::RendererOptions {
+        use_cpu: false,
+        num_init_threads: NonZeroUsize::new(1),
+        antialiasing_support: vello::AaSupport::area_only(),
+    }
+}
+
+impl DrawTarget {
+    fn new(size: Size2D<u32>) -> Self {
+        let mut context = vello::util::RenderContext::new();
+        let device_id = pollster::block_on(context.device(None)).unwrap();
+        let device_handle = &mut context.devices[device_id];
+        let device = device_handle.device.clone();
+        let queue = device_handle.queue.clone();
+        let renderer = vello::Renderer::new(&device, options()).unwrap();
+        let scene = vello::Scene::new();
+        Self {
+            device,
+            queue,
+            renderer: Rc::new(RefCell::new(renderer)),
+            scene,
+            transform: kurbo::Affine::IDENTITY,
+            size,
+        }
+    }
+}
+
+impl GenericDrawTarget<VelloBackend> for DrawTarget {
+    fn clear_rect(&mut self, rect: &Rect<f32>) {
+        // TODO: verify
+        self.scene
+            .push_layer(peniko::Compose::Clear, 0.0, self.transform, &rect.convert());
+        self.scene.fill(
+            peniko::Fill::NonZero,
+            self.transform,
+            peniko::BrushRef::Solid(peniko::color::AlphaColor::TRANSPARENT),
+            None,
+            &rect.convert(),
+        );
+        self.scene.pop_layer();
+    }
+
+    fn copy_surface(&mut self, surface: Vec<u8>, source: Rect<i32>, destination: Point2D<i32>) {
+        todo!()
+    }
+
+    fn create_path_builder(&self) -> kurbo::BezPath {
+        kurbo::BezPath::new()
+    }
+
+    fn create_similar_draw_target(&self, size: &Size2D<i32>) -> Self {
+        Self {
+            device: self.device.clone(),
+            queue: self.queue.clone(),
+            renderer: self.renderer.clone(),
+            scene: vello::Scene::new(),
+            transform: kurbo::Affine::IDENTITY,
+            size: size.cast(),
+        }
+    }
+
+    fn create_source_surface_from_data(&self, data: &[u8]) -> Option<Vec<u8>> {
+        Some(data.to_vec())
+    }
+
+    fn draw_surface(
+        &mut self,
+        surface: Vec<u8>,
+        dest: Rect<f64>,
+        source: Rect<f64>,
+        filter: Filter,
+        draw_options: &DrawOptions,
+    ) {
+        todo!()
+    }
+
+    fn draw_surface_with_shadow(
+        &self,
+        surface: Vec<u8>,
+        dest: &Point2D<f32>,
+        color: &peniko::Color,
+        offset: &Vector2D<f32>,
+        sigma: f32,
+        operator: peniko::BlendMode,
+    ) {
+        todo!()
+    }
+
+    fn fill(&mut self, path: &kurbo::BezPath, pattern: peniko::Brush, _draw_options: &DrawOptions) {
+        self.scene
+            .fill(peniko::Fill::NonZero, self.transform, &pattern, None, path);
+    }
```

Comment:
for each provided draw options we might need to push new layer with draw options then pop it (this should only be needed if draw options are different then current global options, which can happen in backend processing).